.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,93 @@
+
+on:
+  pull_request: ~
+  push:
+    branches:
+      - "main"
+      - "[0-9].*"
+
+name: Static analysis
+
+jobs:
+  phpstan:
+    name: PHPStan
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version: [7.4]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@2.7.0
+        with:
+          coverage: none
+          extensions: intl
+          php-version: ${{ matrix.php-version }}
+
+      - name: Determine composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=directory::$(composer config cache-dir)"
+
+      - name: Cache dependencies installed with composer
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.directory }}
+          key: composer-${{ matrix.php-version }}-composer-${{ hashFiles('composer.*') }}
+          restore-keys: |
+            composer-${{ runner.os }}-${{ matrix.php-version }}-
+            composer-${{ runner.os }}-
+            composer-
+
+      - name: Download dependencies
+        run: composer update --no-interaction --prefer-dist --optimize-autoloader
+
+      - name: PHPStan
+        uses: docker://oskarstark/phpstan-ga:0.12.49
+        with:
+          entrypoint: /composer/vendor/bin/phpstan
+          args: analyze --no-progress
+
+  psalm:
+    name: Psalm
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version: [7.4]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP with extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: none
+          extensions: intl
+          php-version: ${{ matrix.php-version }}
+
+      - name: Determine composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=directory::$(composer config cache-dir)"
+
+      - name: Cache dependencies installed with composer
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.directory }}
+          key: composer-${{ matrix.php-version }}-composer-${{ hashFiles('composer.*') }}
+          restore-keys: |
+            composer-${{ runner.os }}-${{ matrix.php-version }}-
+            composer-${{ runner.os }}-
+            composer-
+
+      - name: Install dependencies with composer
+        run: |
+          composer require --dev psalm/phar:3.18.2 --no-update
```

Comment:
```suggestion
          composer require --dev psalm/phar:^3.18.2 --no-update
```

Reply:
This one is intentional. I dont want to update psalm. That could lead to CI failures for unrelated changes. Ie, CI is red when we change a typo in readme just because we updated psalm.

Reply:
Fair enough.